### PR TITLE
feat: use .NET 6 in lambdas

### PR DESCRIPTION
**Pull request by sobolk**
#### Description of changes

This PR bumps .NET to 6 is lambda runtime and templates.
Existing .NET Core 3.1 functions keep working and cli emits warning.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jhockett — DISMISSED** [review]

File: packages/amplify-dotnet-function-runtime-provider/amplify-plugin.json [hunk 1]
```diff
@@ -11,8 +11,8 @@
     },
     "runtimes": [
       {
-        "name": ".NET Core 3.1",
-        "value": "dotnetcore3.1"
+        "name": ".NET 6",
```
Comment (akshbhu) [inline packages/amplify-dotnet-function-runtime-provider/amplify-plugin.json:14]: Can we support both runtimes and can detect runtime based on Cx project ?

Also we can warn to Cx that ```NET Core 3.1``` is getting deprecated.

Comment (sobolk) [reply packages/amplify-dotnet-function-runtime-provider/amplify-plugin.json:14]: How about this ?
image

Comment (sobolk) [reply packages/amplify-dotnet-function-runtime-provider/amplify-plugin.json:14]: btw. this `amplify-plugin.json`'s `runtimes` section doesn't seem to be used for anything. I'd keep just .net6 here and not advertise deprecated tech. @jhockett @akshbhu do you see any reasons we should keep net core here? we're removing ability to generate new functons from netcore templates.

Comment (akshbhu) [reply packages/amplify-dotnet-function-runtime-provider/amplify-plugin.json:14]: yeah I agree , it's getting deprecated , we shouldn't allow to create ```dotnetcore31 ``` runtime.

Comment (sobolk) [reply packages/amplify-dotnet-function-runtime-provider/amplify-plugin.json:14]: rebased to dev and added migration test.

**Review by jhockett — APPROVED** [review]

**Review by akshbhu — APPROVED** [review]

**Pull request merged**